packages/blade/src/components/Card/CardFooter.tsx

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable @typescript-eslint/no-unnecessary-type-assertion */
+import { Divider } from './Divider';
+import { useInsideCard } from './CardContext';
+import Box from '~components/Box';
+import type { ButtonProps } from '~components/Button';
+import { Button } from '~components/Button';
+import { Text } from '~components/Typography';
+import { useBreakpoint } from '~utils';
+import { useTheme } from '~components/BladeProvider';
+
+export type CardFooterAction = Pick<
+  ButtonProps,
+  'type' | 'accessibilityLabel' | 'isLoading' | 'isDisabled' | 'icon' | 'iconPosition' | 'onClick'
+> & {
+  text: string;
+};
+
+type CardFooterProps = {
+  children?: React.ReactNode;
+};
+
+const useIsMobile = (): boolean => {
+  const { theme } = useTheme();
+  const { matchedDeviceType } = useBreakpoint({
+    breakpoints: theme.breakpoints,
+  });
+  return matchedDeviceType === 'mobile';
+};
+
+const CardFooter = ({ children }: CardFooterProps): React.ReactElement => {
+  const isMobile = useIsMobile();
+  useInsideCard('CardFooter');
+
+  return (
+    <Box marginTop="spacing.7">
+      <Divider />
+      <Box
+        marginTop="spacing.7"
+        display="flex"
+        flexDirection={isMobile ? 'column' : 'row'}
+        justifyContent="space-between"
+      >
+        {children}
+      </Box>
+    </Box>
+  );
+};
+
+type CardFooterLeadingProps = {
+  title: string;
+  subtitle?: string;
+};
+const CardFooterLeading = ({ title, subtitle }: CardFooterLeadingProps): React.ReactElement => {
+  useInsideCard('CardFooterLeading');
+
+  return (
+    <Box>
+      <Text variant="body" size="medium" weight="bold">
+        {title}
+      </Text>
+      <Text variant="body" size="small" weight="regular">
+        {subtitle}
+      </Text>
+    </Box>
+  );
+};
+
+type CardFooterTrailingProps = {
+  actions?: {
+    primary?: CardFooterAction;
+    secondary?: CardFooterAction;
+  };
+};
+const CardFooterTrailing = ({ actions }: CardFooterTrailingProps): React.ReactElement => {
+  const isMobile = useIsMobile();
+  useInsideCard('CardFooterTrailing');
+
+  return (
+    <Box
+      display="flex"
+      flexDirection="row"
+      alignSelf={isMobile ? 'auto' : 'center'}
+      marginTop={isMobile ? 'spacing.5' : 'spacing.0'}
+      marginLeft={isMobile ? 'spacing.0' : 'spacing.5'}
+    >
+      <Box flexGrow={1}>
+        {actions?.secondary ? (
+          <Button isFullWidth size="medium" variant="secondary" {...actions.secondary}>
```

Comment:
Need it so that the buttons span over the entire footer. 

<img width="610" alt="image" src="https://user-images.githubusercontent.com/35374649/208100634-b67b8f78-a081-47a0-9381-e2a2565c07b2.png">

This change was done as part of the responsive issues which I was discussion with Rastogi.

Reply:
but this should maybe apply in case of just mobile and not desktop? because on desktop it might look very big and weird? the below looks ideal IMO?
<img width="629" alt="Screenshot 2022-12-19 at 1 24 24 PM" src="https://user-images.githubusercontent.com/11384858/208374893-03223ba3-498c-498f-85d3-cc9eb466c9bb.png">

Reply:
In desktop it will look as expected. 

<img width="830" alt="image" src="https://user-images.githubusercontent.com/35374649/208608660-3857cfb1-ab73-4c2e-b4ea-4af29dbdef1f.png">

If you notice the Button is wrapped in Box which has `flexGrow={1}` 
The reason I added isFullWidth to Button so that it grows and shrinks with the responsiveness of the Flex items. and on desktop I've added `isMobile ? 'column' : 'row'` thus it won't span the whole card as you've mentioned on desktop.